### ISSUE TITLE
number of clusters

### ISSUE THREAD

**Issue by thomascong121** [shikiw/DAM-VP#6]
Hi, thanks for the great work. I realise that you are using AgglomerativeClustering from sci-kit learn to first learn the cluster prototypes. However, in my case, this algorithm keeps giving me a single cluster, do you have any suggestions on this behavior?

**Comment by shikiw** [comment]
Thanks for your interest!

You can try to configure the threshold in:
```
            hc = cluster.AgglomerativeClustering(
                n_clusters=None, 
                linkage='average', 
                distance_threshold=threshold
            )
```
or, you can assign the ```n_clusters``` if you want to specify the number of clusters.

**Comment by thomascong121** [comment]
Thanks for the reply, I tried your suggestion and it works! 

I do have another question about task-adapting training of the prompts. I realise that you are using the trained meta-prompts as prompt initialisation for task-adapting training, but I didn't see any codes that attempt to save the best prompt in `adapter.py`. Please correct me if I am looking for the wrong codes or you are doing this on-purpose.

**Comment by shikiw** [comment]
Yes, there is no specific code for saving the best prompt. You can add it if needed :)
